src/main/lombok/eu/coinform/gateway/controller/CheckController.java

```diff
@@ -129,6 +130,33 @@ public void corsHeadersResponse(HttpServletResponse response, @PathVariable(valu
         response.addHeader("Access-Control-Max-Age", "3600");
     }
 
+    @CrossOrigin(origins = "*")
+    @JsonView(Views.Debug.class)
+    @RequestMapping(value = "/response/{query_id}/{debug}", method = RequestMethod.GET)
+    public QueryResponse findById(@PathVariable(value = "query_id", required = true) String query_id, @PathVariable(value = "debug", required = true) String debug) {
+
+        log.trace("query for response received with query_id '{}'", query_id);
+
+        QueryResponse queryResponse = redisHandler.getQueryResponse(query_id).join();
+
+        log.debug("findById: {}", queryResponse);
+
+        return queryResponse;
+    }
+
+    //@CrossOrigin("*")
+    @JsonView(Views.Debug.class)
+    @RequestMapping(value = "/response/{query_id}/{debug}", method = RequestMethod.OPTIONS)
+    public void corsHeadersResponse(HttpServletResponse response,
+                                    @PathVariable(value = "query_id", required = true) String query_id,
+                                    @PathVariable(value = "debug", required = true) String debug) {
+        //response.addHeader("Access-Control-Allow-Origin", "https://twitter.com, chrome://**, chrome-extension://**");
+        response.addHeader("Access-Control-Allow-Origin", "*");
+        response.addHeader("Access-Control-Allow-Methods", "GET, OPTIONS");
+        response.addHeader("Access-Control-Allow-Headers", "origin, content-type, accept, x-requested-with");
+        response.addHeader("Access-Control-Max-Age", "3600");
+    }
+
```

Comment:
Why is debug a pathvariable, isn't it nicer to have it set fixed to "debug"

Reply:
Because java cant see the method as disparate from the other findById method which then would have the exact same signature